src/accountsdb/fuzz.zig

```diff
@@ -129,9 +131,25 @@ pub fn run(seed: u64, args: *std.process.ArgIterator) !void {
         manager_handle.join();
     }
 
-    var tracked_accounts = std.AutoArrayHashMap(Pubkey, TrackedAccount).init(allocator);
-    defer tracked_accounts.deinit();
-    try tracked_accounts.ensureTotalCapacity(10_000);
+    const Map = std.AutoArrayHashMap(Pubkey, TrackedAccount);
+    var tracked_accounts_rw = sig.sync.RwMux(Map).init(blk: {
+        var tracked_accounts = Map.init(allocator);
+        try tracked_accounts.ensureTotalCapacity(10_000);
+        break :blk tracked_accounts;
+    });
+    var map_alive = std.atomic.Value(bool).init(true);
+
+    var threads: [N_RANDOM_THREADS]std.Thread = undefined;
+    var spawned_threads: u8 = 0;
+
+    defer {
+        map_alive.store(false, .seq_cst);
+        const tracked_accounts, var tracked_accounts_lg = tracked_accounts_rw.writeWithLock();
+        defer tracked_accounts_lg.unlock();
+        tracked_accounts.deinit();
+
+        for (threads[0..spawned_threads]) |thread| thread.join();
```

Comment:
Should prob join threads outside the writeLock to avoid a deadlock: `reader misses alive, goes to readLock; writer alive=false, writeLocks, waits on reader to join; reader readLock is waiting on writer writeLock`

Maybe move the `defer { alive=false; join threads }` after the `defer deinit`. Could also get rid of the `errdefer detach(); spawned = 0` down below.

Reply:
c8250b5b